### ISSUE TITLE
Regression in parsing wildcard functions

### ISSUE THREAD

**Issue by liufengyun** [scalameta/scalameta#561]
I just upgraded to 1.4.0 and ScalaMeta is no longer able to parse the following valid code:

```Scala
val f: Int => String = _.toString
```

```
error: unbound placeholder parameter
```

**Comment by xeno-by** [comment]
@cb372 Looks like this regression is caused by https://github.com/scalameta/scalameta/commit/070b9eb84507d82f39f364399b8fd3b549a93ce3. Could you take a look?

**Comment by olafurpg** [comment]
Whoops, I need to be more careful reviewing. In fact, I think a test suite parsing millions of lines of Scala code would have caught this regression. I opened #562.

I can also take a look at this @liufengyun if you like?

**Comment by liufengyun** [comment]
Thanks @olafurpg , that's very nice of you to take a look at this, I'm now focusing on another Dotty issue.

**Comment by xeno-by** [comment]
Fixed in 1.5.0.563.
